### PR TITLE
Implement cells, rows, cellIndex IDL attributes and insertRow, deleteRow, insertCell, deleteCell, and cellIndex IDL methods for tabular data

### PR DESCRIPTION
**Pull request by dzbarsky**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6936)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Having this all squashed into one commit makes it hard to figure out if they all have sufficient tests.

**Comment by dzbarsky** [comment]
The issue is that the existing WPT tests depends on all these features, so implementing them separately means none of the tests will pass.  How about I make a list of spec sections/steps that this implements and for each one point out which test tests it (and add missing ones).  That should make reviewing easier.

**Comment by Ms2ger** [comment]
Hm, fun. That would probably help, thanks!

**Comment by dzbarsky** [comment]
Split the caption stuff into https://github.com/servo/servo/pull/7063

**Comment by dzbarsky** [comment]
Split createTBody into https://github.com/servo/servo/pull/7089

**Comment by nox** [comment]
Should that one be closed?

**Comment by dzbarsky** [comment]
@nox I'll try to split some other changes out, but some will stay in this PR.

**Comment by dzbarsky** [comment]
Ready for review. I tried to test that cells and rows filter out children elements that aren't of the right type, but I think the html5 parsing algorithm automatically wraps other elements in `<td>` or `<tr><td>` or something so I couldn't actually test that.

**Comment by Ms2ger** [comment]
You may not be able to get there through the HTML parser, but script can certainly do it.

**Comment by dzbarsky** [comment]
Good point, I added tests for that.

**Comment by dzbarsky** [comment]
@Ms2ger review ping?

Also, tidy is claiming that the use statements are not in alphabetical order, but I think they are?
And AFAICT http://mxr.mozilla.org/servo/source/components/layout/wrapper.rs#43 is not in alphabetical order, so I'm pretty confused by this check.

**Comment by dzbarsky** [comment]
Ok I split the use statements into 1 for each line but kept the same order, and test-tidy is happy now. That check must be buggy.

**Comment by dzbarsky** [comment]
@Ms2ger review ping?

**Comment by frewsxcv** [comment]
As per [this comment](https://github.com/servo/servo/pull/7750#issuecomment-143567322), unless @dzbarsky says otherwise I can take ownership of this PR.

**Comment by frewsxcv** [comment]
Going to open these into smaller individual PRs (starting with #7823). No need to keep this open.

File: components/script/dom/webidls/HTMLTableElement.webidl [hunk 1]
```diff
@@ -6,7 +6,7 @@
 // https://www.whatwg.org/html/#htmltableelement
 interface HTMLTableElement : HTMLElement {
            attribute HTMLTableCaptionElement? caption;
-  HTMLElement createCaption();
```
Comment (frewsxcv) [inline components/script/dom/webidls/HTMLTableElement.webidl]: Do you remember why you did this @dzbarsky? For reference: https://html.spec.whatwg.org/multipage/tables.html#htmltableelement

Comment (dzbarsky) [inline components/script/dom/webidls/HTMLTableElement.webidl]: It lets us get rid of the HTMLElementCasts, and it doesn't really matter what type you declare as being returned in the IDL, what matters is the properties/methods on the type actually returned.

Comment (dzbarsky) [inline components/script/dom/webidls/HTMLTableElement.webidl]: We can probably file a spec bug for this if we care

Comment (nox) [inline components/script/dom/webidls/HTMLTableElement.webidl]: I would rather keep the casts for now.

**Pull request closed without merging**